source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,157 @@ string as follow:
+
+      .. note:: Omitting AWS Session Token
+
+         If you aren't using an AWS session token, omit the ``authMechanismProperties``
+         parameter.
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-aws-mongoclientsettings-mongo-credential
+
+      .. code-block:: csharp
+
+         var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalIdentity("<awsKeyId>"), new PasswordEvidence("<awsSecretKey>"));
+             .WithMechanismProperty("AWS_SESSION_TOKEN", "<awsSessionToken>");
+         var client = new MongoClient(mongoClientSettings);
+
+      .. note:: Omitting AWS Session Token
+
+         If you aren't using an AWS session token, omit the ``WithMechanismProperty()``
+         method call.
+
+Retrieve Credentials Automatically
+++++++++++++++++++++++++++++++++++
+
+Instead of specifying your AWS IAM credentials in ``MongoClientSettings``, you can 
+instruct the {+driver-short+} to use the AWS SDK to automatically retrieve your 
+credentials from an external source. In order for the driver to automatically
+retrieve your credentials, you must perform the following steps:
+
+1. Specify ``MONGODB-AWS`` as the authentication mechanism
+#. Specify that the authentication database name is external to MongoDB
+#. Set your credentials in the appropriate location 
+
+You can specify the authentication mechanism and database either
+by using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying the ``MONGODB-AWS`` authentication mechanism
+and external authentication database name:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-aws-automatic-connection-string
+      
+      .. code-block:: csharp
+
+         var connectionString = "mongodb+srv://<atlasUri>?authMechanism=MONGODB-AWS&authSource=$external";
+         var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+         var client = new MongoClient(mongoClientSettings);
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-aws-automatic-mongo-credential
+
+      .. code-block:: csharp
+
+         var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalAwsIdentity(), new ExternalEvidence());
+         var client = new MongoClient(mongoClientSettings);
+   
+After you specify the authentication mechanism and database name, you must set
+your credentials in the location appropriate to the credential type.
+
+.. important::
+
+   The {+driver-short+} checks for credentials in the following locations in the order
+   listed here.
+
+You can use an OpenID Connect (OIDC)-compatible **web identity provider** to authenticate
```

Comment:
Just checking whether you got the chance to read this and what your thoughts are on it. While passable without introducing the information first, I think it would improve the speed at which readers could locate the relevant info and confirm whether the information they're looking for is in that paragraph faster.

Reply:
Thanks--I missed this comment and I completely agree.